### PR TITLE
Ib/cmssw 7 3 x/gcc491

### PR DESCRIPTION
**Pull request by smuzaffar**

### CODE REVIEW

File: fwlite.spec [hunk 1]
```diff
@@ -4,15 +4,20 @@ Requires: fwlite-tool-conf python
...
+%define branch          CMSSW_7_0_X
+%define gitcommit       %(echo %realversion | sed -e 's|_FWLITE||')
 
 # Switch off building tests
 %define patchsrc perl -p -i -e ' s|(<classpath.*test\\+test.*>)||;' config/BuildFile.xml*
 
-# Includes parts of the framework that we don't want in fwlite
-%define patchsrc2 rm -rf src/DataFormats/GeometrySurface/plugins
+#patch to build fwlite release; this should be fixed in cmssw
+%define patchsrc2 sed -i -e 's|^#include "Geometry/GEMGeometry/|//#include "Geometry/GEMGeometry/|' src/Fireworks/Muons/plugins/FWGEMDigiProxyBuilder.cc
```
Comment (smuzaffar) [inline fwlite.spec:14]: there are pull requests for 74x/73x to fix this

File: fwlite.spec [hunk 2]
```diff
@@ -4,15 +4,20 @@ Requires: fwlite-tool-conf python
 
 %define useCmsTC        yes
 %define saveDeps        yes
+%define branch          CMSSW_7_0_X
```
Comment (smuzaffar) [inline fwlite.spec:7]: we need to use CMSSW_7_0_X here as CMSSW_7_3_0 release was build with branch=CMSSW_7_0_X
if we change it to CMSSW_7_3_X here then fwlite will again download the sources from github and upload to cmsrep

File: fwlite_build_set.file [hunk 3]
```diff
@@ -115,3 +119,10 @@ SimDataFormats/TrackingHit
 SimDataFormats/Vertex
 Utilities/General
 Utilities/ReleaseScripts
+Geometry/CommonDetUnit
+SimDataFormats/JetMatching/interface/JetFlavourInfo.h
+Geometry/CommonTopologies/interface/Topology.h
+DataFormats/TrackerCommon/interface/TrackerTopology.h
+Geometry/TrackerGeometryBuilder/interface/GluedGeomDet.h
+Geometry/TrackerGeometryBuilder/interface/GeomDetLess.h
+RecoLocalCalo/HcalRecAlgos/interface/HcalCaloFlagLabels.h
```
Comment (smuzaffar) [inline fwlite_build_set.file:128]: these are source dependencies of fwlite, so we only include these files
we should re-think the partial builds may be sub-packages might be a better way?

**Comment by ghost** [comment]
test

**Comment by nclopezo** [comment]
For now it only works for cmssw prs

**Comment by nclopezo** [comment]
I started the tests here: https://cmssdt.cern.ch/jenkins/job/test-externals-prs/39/console

**Pull request merged**